src/event-manager.js

```diff
@@ -67,49 +67,53 @@ class CapturedHandlerEntry {
   }
 }
 
-function handleDelegatedEvent(event) {
-  event.propagationStopped = false;
-  let target = findOriginalEventTarget(event);
+class DelegateHandlerEntry {
+  constructor(eventName, eventManager) {
+    this.eventName = eventName;
+    this.count = 0;
+    this.eventManager = eventManager;
+  }
 
-  while (target && !event.propagationStopped) {
-    if (target.delegatedCallbacks) {
-      let callback = target.delegatedCallbacks[event.type];
-      if (callback) {
-        if (event.stopPropagation !== stopPropagation) {
-          event.standardStopPropagation = event.stopPropagation;
-          event.stopPropagation = stopPropagation;
-        }
-        if ('handleEvent' in callback) {
-          callback.handleEvent(event);
-        } else {
-          callback(event);
+  handleDelegatedEvent(event) {
```

Comment:
```suggestion
  handleEvent(event) {
```
And then in `increment()`:
```ts
DOM.addEventListener(this.eventName, this, false);
```
and in `decrement()`:
```ts
DOM.removeEventListener(this.eventName, this, false);
```

Reply:
@bigopon Updated the PR. Thanks for this reply! This resolved an issue I could not wrap my mind around, I thought something was wrong with the unit tests cleaning the listeners but apparently I just made a mistake in the code itself... nice catch